src/mocha/utils.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+'use strict';
+
+/* eslint-env browser, worker, mocha */
+
+const mochaUtils = {
+  startMochaTest: () => {
+    let passedTests = [];
+    let failedTests = [];
+
+    mocha.checkLeaks();
+
+    var runResults = window.mocha.run();
```

Comment:
You've got some conditionals lower down (`typeof window !== 'undefined'`...) which detect whether this code is being run from an environment in which `window` is defined or not, but here you're assuming that `window.mocha` will be defined.